homeassistant/components/recorder/__init__.py

```diff
@@ -94,7 +94,7 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-T = TypeVar("T")
+T = TypeVar("T")  # pylint: disable=invalid-name
```

Comment:
With the new pylint version, `T` is an allowed name now. So `disable=invalid-name` shouldn't be necessary. #68656

Besides that, I would recommend using a private name (i.e. `_T`) for TypeVars. See my comment here: https://github.com/home-assistant/core/pull/68205#issuecomment-1072548688

Reply:
It would definitely be good to switch it over in the future once we get the rest of the typing done 👍